### PR TITLE
De353042 network reachability

### PR DESCRIPTION
**Pull request by jhoongo**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,11 @@
 ### Enhancement
 Swift code is added to the [iOS Guide](http://mas.ca.com/docs/ios/latest/guides/). The iOS API Reference documentation is based on Objective-c, but the Objective-c code can be used seamlessly with your Swift code. See the usage section for details.
 
+### New features
+- `[MAS setNetworkMonitorBlockForHost:monitoringBlock:]` and `[MAS isNetworkReachableForHost:]` are added to monitor a specific host's network reachability through Mobile SDK. [DE353042]  
+
```
Comment (bmckercher) [inline CHANGELOG.md]: +- `[MAS setNetworkMonitorBlockForHost:monitoringBlock:]` and `[MAS isNetworkReachableForHost:]` are added to monitor a specific host's network reachability through the Mobile SDK. [DE353042]

Should we add that this was added to MASFoundation for clarity? Your call.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,11 @@
...
 ### Bug fixes
+- `[MAS setGatewayMonitor:]`, `[MAS gatewayIsReachable]`, and `[MAS gatewayMonitoringStatusAsString]` were providing results of device's internet connectivity.  The Mobile SDK now returns network reachability of the primary gateway for those methods. [DE353042]
```
Comment (bmckercher) [inline CHANGELOG.md]: +- `[MAS setGatewayMonitor:]`, `[MAS gatewayIsReachable]`, and `[MAS gatewayMonitoringStatusAsString]` were providing results of the device's internet connectivity.  The Mobile SDK now returns network reachability of the primary gateway for these methods. [DE353042]

File: MASFoundation/Classes/MAS.h [hunk 3]
```diff
@@ -337,6 +318,27 @@
...
+ *
+ *  The gateway monitoring status enumerated values are:
+ *
+ *      MASGatewayMonitoringStatusNotReachable
+ *      MASGatewayMonitoringStatusReachableViaWWAN
+ *      MASGatewayMonitoringStatusReachableViaWiFi
+ *
+ *  This is optional and it can be set to nil at any time to stop receiving the notifications.
+ *
+ *  Reachability status update can also be received as NSNotification.  Subscribe notification key, MASGatewayMonitorStatusUpdateNotification, and notifications will be broadcasted whenever there is a change in network reachability status for the primary gateway on the plaload of NSNotification as NSDictionary format @{ (NSString)"host" : (NSNumber)enumValue}:
```
Comment (bmckercher) [inline MASFoundation/Classes/MAS.h]: *  Reachability status update can also be received as NSNotification.  Subscribe notification key, MASGatewayMonitorStatusUpdateNotification, and notifications will be broadcasted whenever there is a change in network reachability status for the primary gateway on the payload of NSNotification as NSDictionary format @{ (NSString)"host" : (NSNumber)enumValue}:

File: MASFoundation/Classes/MAS.h [hunk 4]
```diff
@@ -361,6 +363,40 @@
...
+ 
+    MASNetworkReachabilityStatusUnknown
+    MASNetworkReachabilityStatusNotReachable
+    MASNetworkReachabilityStatusReachableViaWWAN
+    MASNetworkReachabilityStatusReachableViaWiFi
+    MASNetworkReachabilityStatusInitializing
+ 
+ This is optional and it can be set to nil at any time to stop receiving the notifications.
+ 
+ Reachability status update can also be received as NSNotification.  Subscribe notification key, MASNetworkReachabilityStatusUpdateNotification, and notifications will be broadcasted whenever there is a change in network reachability status for the specified host on the plaload of NSNotification as NSDictionary format @{ (NSString)"host" : (NSNumber)enumValue}:
```
Comment (bmckercher) [inline MASFoundation/Classes/MAS.h]: same as above "payload"

File: MASFoundation/Classes/_private_/services/network/internal/MASNetworkReachability.h [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+///--------------------------------------
+/// @name Lifecycle
+///--------------------------------------
+
+# pragma mark - Lifecycle
+
+
+/**
+ Initializer to perform defualt object initialization with specified domain value as in string.
```
Comment (bmckercher) [inline MASFoundation/Classes/_private_/services/network/internal/MASNetworkReachability.h]: "default"

File: MASFoundation/Classes/_private_/services/network/internal/MASNetworkReachability.h [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+/**
+ Starts monitoring network reachability status for given domain.
+ */
+- (void)startMonitoring;
+
+
+
+/**
+ Stops monitoring network reachability status for given domain.  If the object is already monitoring the reachability, it will stop monitoring first, then re-start.
```
Comment (bmckercher) [inline MASFoundation/Classes/_private_/services/network/internal/MASNetworkReachability.h]: "restart"

**Review by bmckercher — DISMISSED** [review]
Very minor edits. Looks good!

**Review by rodrigoareis — APPROVED** [review]
looks good to me

**Comment by ghost** [comment]
Result of [Integration 1](https://stlt.herokuapp.com/v1/xcs_deeplink/qa-mas-mac-mini-1.ca.com/df998acc0587b1333d4f704f04c12338/df998acc0587b1333d4f704f04c1469c)
---
*Duration*: 3 minutes and 52 seconds
*Result*: All 0 tests passed, but please **fix 35 warnings** and **19 analyzer warnings**.

**Pull request merged**